### DISCUSSION TITLE
upload file via zosmf could fail

### DISCUSSION THREAD

**Discussion by kraabrsg** [zowe/zowe-cli#2227] [Q&A]
Hi we are using the sdk in conjunction with **ZOSMF** in our vscode extension

        "@zowe/zos-files-for-zowe-sdk": "^7.26.1",
        "@zowe/zos-jobs-for-zowe-sdk": "^7.26.1",
        "@zowe/zos-tso-for-zowe-sdk": "^7.25.2",
        "@zowe/zosmf-for-zowe-sdk": "^7.25.2",

**Whats the basic implementation ?**
1.) make folders in vscode for the current user if they don'T exists
2.) list members of the PO Dataset on the Mainframe
3.) sync these folders local, with Mainframe as master
4.) Use DOWNLOAD.ALLMEMBERS to get the content

**When a user opens a file in vscode**
 `Download.dataSet `
is called (with async await) .

**When the user saves this is called (wrapped in async function)**
`  response = await Upload.fileToDataset(this.session, inputFile, this.encodeDatasetName(hostDsName), { encoding: Configuration.hostEncoding, responseTimeout: 5000 });`
   

The Problem is now when the Mainframe is really busy it could happen that users end up with truncated or empty files. 
Are there any tips to avoid this, or are there any reports from Users with the same problem. Maybe configuration tips for zosmf or something similar (We also have checks if the file is different local vs. Mainframe).

Thank you!

**Comment by awharn** [comment]
Hello. The team is not aware of any other users reporting this issue. The REST client should be throwing an error if the server responds with any error code in the 4XX-5XX range, which should occur if z/OSMF encounters an error while uploading the file contents to the data-set. It is possible this may be a bug with z/OSMF itself, or the wrapped asynchronous function may not be properly awaited. If the upload issue happens again, please check your z/OS syslog and z/OSMF job log to see if they contain any error messages that may be relevant to help troubleshoot the issue.
